### PR TITLE
vic-ui allows HTTPS plugin URL

### LINKED ISSUE #2045: --serverThumbprint flag is not recognized in vic-ui
Using build 4031, UI installer fails because `vic-ui` it couldn't recognize the `--serverThumbprint` flag which is needed for plugin deployment 

**Expected**
`vic-ui install ~~~~ --serverThumbprint 11 22 33 AA BB CC ...` should complete successfully

**Actual**
It fails with the following message

```
kjosh-m03:VCSA kjosh$ ./install.sh 
Enter your vCenter Administrator Username: administrator@vsphere.local
Enter your vCenter Administrator Password: 
Reading plugin-package.xml...
----------------------------------------
Registering vCenter Server Extension...
----------------------------------------
../../vic-ui-darwin install --target https://10.160.3.71/sdk/ --user administrator@vsphere.local --password Ca$hc0wvc --serverThumbprint 8C C0 E5 13 D5 0F 81 41 E5 1A E8 B0 BE 26 1C 65 69 93 81 1F --key com.vmware.vicui.Vicui --name VicUI --version 0.0.1 --summary VicUI --company VMware --url https://10.12.35.129:3443/vsphere-plugins/com.vmware.vicui.Vicui-0.0.1.zip
Incorrect Usage.

NAME:
   vic-ui-darwin install - Install UI plugin

USAGE:
   vic-ui-darwin install [command options] [arguments...]

OPTIONS:
   --target value, -t value    ESXi or vCenter connection URL, specifying a datacenter if multiple exist e.g. root:password@VC-FQDN/datacenter (default: https://10.160.3.71/sdk/)
   --user value, -u value      ESX or vCenter user
   --password value, -p value  ESX or vCenter password (default: Ca$hc0wvc)
   --force, -f                 Force install
   --company value             Plugin company name (required)
   --key value                 Plugin key (required)
   --name value                Plugin name (required)
   --no-show                   Hide plugin in UI
   --summary value             Plugin summary (required)
   --url value                 Plugin URL (required)
   --version value             Plugin version (required)
   --debug value, -v value     [0(default),1...n], 0 is disabled, 1 is enabled, >= 1 may alter behaviour (default: 0)
   
ERRO[2016-08-18T13:22:11-05:00] --------------------                         
ERRO[2016-08-18T13:22:11-05:00] vic-ui-darwin failed: flag provided but not defined: -serverThumbprint
 
-------------------------------------------------------------
Error! Could not register plugin with vCenter Server. Please see the message above
kjosh-m03:VCSA kjosh$ 
```

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #2045

### CODE REVIEW

**Comment by jooskim** [comment]
LGTM

File: cmd/vic-ui/ui/ui.go [hunk 1]
```diff
@@ -137,6 +145,10 @@ func (p *Plugin) processInstallParams() error {
 		return cli.NewExitError("--version must be specified", 1)
 	}
 
+	if strings.HasPrefix(strings.ToLower(p.URL), "https://") && p.ServerThumbprint == "" {
```
Comment (emlin) [inline cmd/vic-ui/ui/ui.go]: URL.Scheme should be "https" in your case

File: lib/install/plugin/register.go [hunk 2]
```diff
@@ -150,7 +152,18 @@ func (p *Pluginator) Register() error {
 
 	e.ResourceList = append(e.ResourceList, eri)
 
-	// TODO HTTP server info
+	// HTTPS requires extension server info
+	if strings.HasPrefix(strings.ToLower(p.info.URL), "https://") {
```
Comment (emlin) [inline lib/install/plugin/register.go]: ditto^

**Comment by emlin** [comment]
Just curious if thumbprint is wrong, will user get error message during registration?

**Comment by jooskim** [comment]
It won't, but Web Client will fail when trying to fetch it

**Comment by emlin** [comment]
That is my concern. But it should be fine at this time, as it's not end user facing command line.

**Comment by jooskim** [comment]
Yeah, but the other end on the vSphere API doesn't seem to do any validation on it. It'd be nice though if we had this kind of check before going ahead and registering the plugin.

File: cmd/vic-ui/ui/ui.go (hunk 1 above)
Comment (andrewtchin) [inline cmd/vic-ui/ui/ui.go]: `ExtensionServerInfo` takes a string so URL here isn't url.URL

**Comment by emlin** [comment]
thought the URL is url.URL type. So skip that comment at this time, as vsphere api requires string parameter.
LGTM

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**